src/util/rangelist.cpp

```diff
@@ -35,14 +36,13 @@ QList<int> parseRangeList(const QString& input) {
             endIndex = startIndex;
         }
         for (int currentIndex = startIndex; currentIndex <= endIndex; currentIndex++) {
-            intList.append(currentIndex);
+            // Add values sorted and skip duplicates
+            auto insertPos = std::lower_bound(intList.begin(), intList.end(), currentIndex);
+            if (insertPos == intList.end() || *insertPos != currentIndex) {
+                intList.insert(insertPos, currentIndex);
+            }
```

Comment:
technically quite more inefficient because the insert is not constant time but ill let it slide...